roles/pulp_health_check/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+---
+- meta: flush_handlers
+
+- name: Get services
+  service_facts:
+
+- name: Listing Pulp services
+  debug:
+    msg: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+
+- name: Checking Pulp services
+  fail:
+    msg: "{{ item.value.name }} state: {{ item.value.state }}"
+  when: "item.value.name != 'pulpcore-worker@.service' and item.value.state != 'running'"
+  with_dict: '{{ ansible_facts.services|dict2items|selectattr("key", "contains", "pulp")|list|items2dict }}'
+
+- name: Ensure Pulp is up and healthy
+  uri:
+    url: "http://{{ pulp_api_bind }}/pulp/api/v3/status/"
+    user: admin
```

Comment:
Do we have the ability to customize the admin username somewhere else? If that is the case then we might need to use the same var here.

Reply:
We do not. I asked him to hardcode it for that reason. If we made it configurable in this role that currently corresponds to pulpcore 3.5.1 / 3.6.0 specifically, users would assume pulpcore 3.5 /1 3.6.0 supports changing it.